flask/README.md

```diff
@@ -1,24 +1,124 @@
+
+### Set-up pre-commit
+
+[Pre-commit](https://pre-commit.com/) enables us to run code quality checks, such as missing semicolons, trailing whitespace, and debug statements, before you are committing your code. We chose pre-commit since it enables us to run these checks for both frontend and backend in just one place.
+Please follow these steps to set-up pre-commit:
+
+(optional) 0.1 If you have not already done it, create a venv in which you are running pre-commit. If you are using pyenv you might want to check which python version you are using with `pyenv version` or `which python`.
 
     python3 -m venv .venv
+
+(optional) 0.2 Activate the virtual environment. If you are not using direnv, be aware that you should do this step each time before working with python in the Boxtribute repo.
+
     source .venv/bin/activate
-    pip install -e . -r requirements-dev.txt
+
+1. Install pre-commit and the linters/formatters (all declared in `/flask/requirements-dev.txt`)
+
+    pip install -e flask -r flask/requirements-dev.txt
+
+2. Install the hooks to run pre-commit before you commit.
+
     pre-commit install --overwrite
 
 Now you're all set up using Python code quality tools! `pre-commit` automatically checks the staged patch before committing. If it rejects a patch, add the corrections and try to commit again.
 
-Whenever you start a new shell to run tests, style-checks, or work on some code in general, activate the Python virtual environment
+To figure out what else you can do with pre-commit, check out this  [link](https://pre-commit.com/#usage).
 
-    source .venv/bin/activate
+### Linting and Formatting in VSCode
+
+Most of our developers are using VSCode. Instead of running our linter (flake8) and our formatter (black) for python just when you are committing your code, we added a few settings in `.vscode/settings.json` so that your files are formatted and linted when you save a python file. You might want to check out this settings file.
+
+### Working with MySQL
 
-Have a look at [direnv](https://github.com/direnv/direnv) if you're interested in ways to automate this procedure. There is already a `.envrc` file in the root of this repo. If you install `direnv` and allow to run it for your local repo, it will access the python virtual environment `venv` everytime you enter the folder via a command line.
+Since we are working with docker you do not have to install a local MySQL server on your computer. Instead, you can just connect to the MySQL server in one of the Docker containers.
 
-### Testing
+#### General notes on Docker network
 
-#### Writing tests
+In the docker-compose file we define a separate docker network called `backend` to which the backend containers are joined. Each container can now look up the hostname `flask` or `mysql` and get back the appropriate container’s IP address.
+To access the mysql database, there are now two possibilities:
+1. You reach the mysql db at `MYSQL_HOST=mysql` and `MYSQL_PORT=3306` or
```

Comment:
By this you mean the dev should use the mysql command line client on their host machine?

Reply:
I just copy pasted this section from the general README to the flask README. It has nothing to do with wether you want to use the command line client or workbench. Just how to reach the mysql container in your network